client/layout/masterbar/logged-in.jsx

```diff
@@ -201,140 +223,272 @@ class MasterbarLoggedIn extends Component {
 		);
 	}
 
-	render() {
-		const isWordPressActionSearchFeatureEnabled = config.isEnabled( 'wordpress-action-search' );
-		const {
-			domainOnlySite,
-			translate,
-			isCheckout,
-			isCheckoutPending,
-			isMigrationInProgress,
-			previousPath,
-			siteSlug,
-			isJetpackNotAtomic,
-			title,
-			currentSelectedSiteSlug,
-			currentSelectedSiteId,
-		} = this.props;
+	handleToggleMenu = () => {
+		this.setState( { isMenuOpen: ! this.state.isMenuOpen } );
+	};
 
-		const { isActionSearchVisible } = this.state;
+	dismissPopover = () => {
+		this.props.savePreference( MENU_POPOVER_PREFERENCE_KEY, true );
+	};
 
-		if ( isCheckout || isCheckoutPending ) {
-			return (
+	checkout = () => {
+		const { isCheckoutPending, previousPath, siteSlug, isJetpackNotAtomic, title } = this.props;
+		return (
+			<AsyncLoad
+				require="calypso/layout/masterbar/checkout.tsx"
```

Comment:
Do we really need the `.tsx`? This was like that before, so feel free to leave it like that.